[Serverless][8.16] Logs request during preview rule execution (#5871)
elastic/security-docs: docs/detections/rules-ui-create.asciidoc

* To close the preview, click the *Rule preview* button again.

[discrete]
[[debug-rule-queries]]

> ==== Debug rule queries (optional)

Author: I wanted to make the title goal-oriented, but maybe that's not the right direction, especially since this feature only provides the query for debugging. How about this instead:

```suggestion
==== Inspect {es} queries sent by your rule (optional)
```

> NOTE: This option is only offered for {esql} rules.

Author: ```suggestion
NOTE: This option is only offered for {esql} and event correlation rules.
```

When previewing an {esql} rule, you can also learn more about the {es} queries that are submitted when the rule runs. This information can be helpful for identifying and troubleshooting potential rule issues, or validating that your rule is retrieving the expected data.    

To learn more your rule's {es} queries, preview its results and do the following:

. Beneath the rule preview's date and time picker, find the **Show Elasticsearch requests, ran during rule executions** option and select it. The **Preview logged results** section displays under the the histogram and alerts table in the panel. 
. Expand the **Preview logged results** section to display subsections with more information about the rule's {es} queries. The following details are provided:

> ** The expected start date and time of each rule execution and how long it took to complete

Author: I can remove "expected" here.

```suggestion
** The start date and time for each rule execution and how long it took to complete
```

** A brief explanation of the {es} queries

> ** The actual {es} queries that the rule submits when it runs   

Author: Added in [d9b849c](https://github.com/elastic/security-docs/pull/5871/commits/d9b849c4f26df8d55837521b72ad3d3c0d983b25).

* To close the preview, click the *Rule preview* button again.

[discrete]
[[view-rule-es-queries]]
==== View your rule's {es} queries (optional)

NOTE: This option is only offered for {esql} and event correlation rules.

When previewing a rule, you can also learn about its {es} queries, which are submitted when the rule runs. This information can help you identify and troubleshoot potential rule issues. You can also use it to confirm that your rule is retrieving the expected data.   

To learn more about your rule's {es} queries, preview its results and do the following:

> . Select the **Show Elasticsearch requests, ran during rule executions** option below the preview's date and time picker. The **Preview logged results** section displays under the histogram and alerts table. 

Reviewer: ```suggestion
. Select the **Show {es} requests, ran during rule executions** option below the preview's date and time picker. The **Preview logged results** section displays under the histogram and alerts table. 
```

. Click the **Preview logged results** section to expand it. Within the section, each rule execution is shown on an individual row. 
. Expand each row to learn more about the {es} queries that the rule submits each time it executes. The following details are provided:
** When the rule execution started, and how long it took to complete
** A brief explanation of what the {es} queries do

> ** The actual {es} queries that the rule sends to search indices containing events that are used during alert creation   

Reviewer 2: ```suggestion
** The actual {es} queries that the rule sends to search indices containing events during rule execution  
```

rule execution not always lead to alerts creation, there might not be any alerts

Author: Making one more minor adjustment and re-adding "used" in.

+

> TIP: Run the queries in Dev Tools (**{kib}** -> **Management** -> **Dev Tools**) to determine if your rule is retrieving the expected data. For example, to test your rule’s exceptions, run the rule’s {es} queries, which will also contain exceptions added to the rule. If your rule’s exceptions are working as intended, the query will not return events that should be ignored.

Reviewer: I'd hesitate to include specific nav instructions, considering how the nav varies between classic ESS, solution view ESS (the "new nav"), and serverless. Maybe link to the [ESS docs for Console](https://www.elastic.co/guide/en/kibana/master/console-kibana.html) instead?

Also, I don't think it's necessary to use different names for the same UI ("Dev Tools" in ESS docs, and "Console" in serverless docs) — that's just introducing more granular differences between the two content sets. "Console" seems the best option for both ESS and serverless, since it's more specific than "Dev Tools."

Author: ```suggestion
TIP: Run the queries in {kibana-ref}/console-kibana.html[Console] to determine if your rule is retrieving the expected data. For example, to test your rule’s exceptions, run the rule’s {es} queries, which will also contain exceptions added to the rule. If your rule’s exceptions are working as intended, the query will not return events that should be ignored.
```

> TIP: Run the queries in Dev Tools (**{kib}** -> **Management** -> **Dev Tools**) to determine if your rule is retrieving the expected data. For example, to test your rule's exceptions, use the rule's {es} queries to find events that should be ignored during alert creation. If alerts weren't created for those events, your rule's exceptions are working as expected.

Reviewer 2: ```suggestion
TIP: Run the queries in Dev Tools (**{kib}** -> **Management** -> **Dev Tools**) to determine if your rule is retrieving the expected data. For example, to test your rule's exceptions, use the rule's {es} queries to verify resulted query contains expected exceptions and events ignored as intended.
```

Author: Making some minor adjustments based on what we talked about in Slack:

```suggestion
TIP: Run the queries in Dev Tools (**{kib}** -> **Management** -> **Dev Tools**) to determine if your rule is retrieving the expected data. For example, to test your rule’s exceptions, run the rule’s {es} queries, which will also contain exceptions added to the rule. If your rule’s exceptions are working as intended, the query will not return events that should be ignored.
```
